packages/shared/src/rpc/index.ts

```diff
@@ -0,0 +1,181 @@
+
+  return Object.fromEntries(
+    Object.entries(obj).map(([key, value]) => [
+      camelCase(key),
+      camelCaseKeys(value),
+    ]),
+  ) as CamelCaseRecord<T>;
+};
+
+const RPC_URLS: Record<ChainflipNetwork, string> = {
+  [ChainflipNetworks.backspin]: 'https://backspin-rpc.staging',
+  [ChainflipNetworks.sisyphos]: 'https://sisyphos.chainflip.xyz',
+  [ChainflipNetworks.perseverance]: 'https://perseverance.chainflip.xyz',
+  [ChainflipNetworks.mainnet]: 'https://mainnet.chainflip.io',
+};
+
+export type RpcConfig = { rpcUrl: string } | { network: ChainflipNetwork };
+
+const createRequest =
+  <P extends z.ZodTypeAny, R extends z.ZodTypeAny>(
+    method: string,
+    responseParser: R,
+    paramsParser?: P,
+  ) =>
+  async (
+    urlOrNetwork: RpcConfig,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    params: z.input<P> extends any ? void : z.input<P>,
+  ): Promise<CamelCaseRecord<z.output<R>>> => {
+    const url =
+      'network' in urlOrNetwork
+        ? RPC_URLS[urlOrNetwork.network]
+        : urlOrNetwork.rpcUrl;
+    const { data } = await axios.post(url, {
+      jsonrpc: '2.0',
+      method,
+      params: paramsParser?.parse(params),
+      id: 1,
+    });
+
+    const result = responseParser.safeParse(data.result);
+
+    if (result.success) {
+      return camelCaseKeys(result.data);
+    }
+
+    throw new Error(`RPC request "${method}" failed`, { cause: data.error });
+  };
+
+const fundingEnvironment = z.object({
+  redemption_tax: numberOrHex,
+  minimum_funding_amount: numberOrHex,
+});
+export const getFundingEnvironment = createRequest(
+  'cf_funding_environment',
+  fundingEnvironment,
+);
+
+const chainAssetMap = <Z extends z.ZodTypeAny>(parser: Z) =>
+  z.object({
+    Bitcoin: z
+      .object({ Btc: parser })
+      .transform(({ Btc }) => ({ BTC: Btc as z.output<Z> })),
+    Ethereum: z
+      .object({
+        Eth: parser,
+        Usdc: parser,
+        Flip: parser,
+      })
+      .transform(({ Eth, Usdc, Flip }) => ({
+        ETH: Eth as z.output<Z>,
+        USDC: Usdc as z.output<Z>,
+        FLIP: Flip as z.output<Z>,
+      })),
+
+    Polkadot: z
+      .object({ Dot: parser })
+      .transform(({ Dot }) => ({ DOT: Dot as z.output<Z> })),
+  });
+
+export type ChainAssetMap<T> = {
+  Bitcoin: {
+    BTC: T;
+  };
+  Ethereum: {
+    ETH: T;
+    USDC: T;
+    FLIP: T;
+  };
+  Polkadot: {
+    DOT: T;
+  };
+};
+
+const chainAssetNumberMap = chainAssetMap(numberOrHex);
+
+const swappingEnvironment = z.object({
+  minimum_swap_amounts: chainAssetNumberMap,
+});
+export const getSwappingEnvironment = createRequest(
+  'cf_swapping_environment',
+  swappingEnvironment,
+);
+
+const ingressEgressEnvironment = z.object({
+  minimum_deposit_amounts: chainAssetNumberMap,
+});
+
+export const getIngressEgressEnvironment = createRequest(
+  'cf_ingress_egress_environment',
+  ingressEgressEnvironment,
+);
+
+const rpcAsset = z.union([
+  z.literal('Btc'),
```

Comment:
this should not be returned from the rpc anymore, right?

Reply:
wtf happened